mystery_o_matic/output/latex/utils.py

```diff
@@ -0,0 +1,110 @@
+    return r
+
+def save_tex(outdir, language, tex):
+    try:
+        makedirs(outdir + f"/{language}")
+    except OSError:
+        pass
+
+    filename = outdir + f"/{language}/mystery.tex"
+    with open(filename, "w") as f:
+        f.write(tex)
+
+    return filename
+
+def get_char_name(name):
+    if name == "NOBODY":
+        return name
+    return name.capitalize()
+
+def get_emoji_name(emoji_char):
+    # Use emoji.demojize to get the name in the format ":emoji_name:"
+    emoji_name = demojize(emoji_char)
+    emoji_name = emoji_name.strip(":").replace("_", "-")
+    # Strip the colons to return just the canonical name
+    return "\\emoji{" + emoji_name + "}"
+
+def replace_emojis(text):
+    no_emojis = demojize(text).replace("(:", "(\\emoji{")
+    no_emojis = no_emojis.replace(":)", "})")
+    no_emojis = no_emojis.replace("_", "-")
+    return no_emojis
+
+def generate_latex_clue_table(name, num_columns, num_rows, include_header = True):
+    # Define the LaTeX column specifier string
+    column_spec = '|c|c|' + '|'.join(['>{\\centering}m{0.04\\paperwidth}' for _ in range(num_columns)]) + '|'
+
+    # Start building the LaTeX table code
+    latex_code = "\\begin{tabular}{" + column_spec + "}\n\\hline\n"
+
+    # First row with $$ROOM0REP and times
+
+    multirow_number = num_rows
+    if (include_header):
+        multirow_number += 1
+
+    latex_code += "\\multirow{ "+ str(multirow_number) + "}{*}{ $$" + name + " } "
+
+    if (include_header):
+        latex_code += "& \\emoji{mantelpiece-clock} & "
+        # Add the $$TIME elements in the first row
+        time_cells = [f"$$TIME{i}" for i in range(num_columns)]
+        latex_code += " & ".join(time_cells) + " \\tabularnewline\n"
+
+        # Add individual clines for each column from the second to the last
+        for col in range(2, num_columns + 2):
+            latex_code += f"\\cline{{{col}-{num_columns + 2}}} "
+        latex_code += "\n"
+
+    # Rows with $$CHAR1, $$CHAR2, $$CHAR3
+    for char_num in range(1, num_rows + 1):
+        latex_code += f" & $$CHAR{char_num} & " + " & ".join([""] * num_columns) + " \\tabularnewline\n"
+
+        # Add individual clines for each column in the current row
+        if (char_num < 3):
+            for col in range(2, num_columns + 2):
+                latex_code += f"\\cline{{{col}-{num_columns + 2}}} "
+            latex_code += "\n"
+
+    # End the table
+    latex_code += "\\hline\n\\end{tabular}\\\\\n"
+
+    return latex_code
+
+def generate_latex_weapons_table(numer_weapons):
+    latex_code = '\\begin{tabular}{|>{\centering}p{0.172\paperwidth}|>{\centering}p{0.172\paperwidth}|>{\centering}p{0.172\paperwidth}|>{\centering}p{0.172\paperwidth}|}\n'
+    latex_code += "\\hline\n"
+    for i in range(numer_weapons):
+        latex_code += f"$$ROOM{i}WEAPONREP & "
+
+    latex_code = latex_code[:-2] + " \\tabularnewline\n"
+    latex_code += "\\hline\n\\end{tabular}\n"
+    return latex_code
```

Comment:
The typo was fixed. Regarding the other remark, please add a comment about it in [#14](https://github.com/mystery-o-matic/mystery-o-matic.github.io/issues/14) so it will be fixed in a future PR